### PR TITLE
Clean code and improve code comment in tachyon.client

### PR DESCRIPTION
**Pull request by apc999**
Fixing a bunch of typos and clarifying the comments, refactoring code a little, removed unused code path.

### CODE REVIEW

File: core/src/main/java/tachyon/client/BlockOutStream.java [hunk 1]
```diff
@@ -236,8 +232,8 @@ public void write(int b) throws IOException {
       throw new IOException("Out of capacity.");
     }
 
-    if (mBuffer.position() >= mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES,
-        Constants.MB)) {
+    if (mBuffer.position()
+        >= mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB)) {
```
Comment (hsaputra) [inline core/src/main/java/tachyon/client/BlockOutStream.java]: Why is this one better than before?

File: core/src/main/java/tachyon/client/TachyonFile.java [hunk 2]
```diff
@@ -443,14 +446,15 @@ private TachyonByteBuffer readLocalByteBuffer(int blockIndex, long offset, long
...
-   * 
+   *
    * @param blockInfo The blockInfo of the block to read.
    * @return TachyonByteBuffer containing the block.
    */
   TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) {
     // We call into the remote block in stream class to read a remote byte buffer
-    ByteBuffer buf = RemoteBlockInStream.readRemoteByteBuffer(mTachyonFS,
-        blockInfo, 0, blockInfo.length, mTachyonConf);
+    ByteBuffer buf =
```
Comment (hsaputra) [inline core/src/main/java/tachyon/client/TachyonFile.java]: Why is this format better than before?

Comment (apc999) [inline core/src/main/java/tachyon/client/TachyonFile.java]: This was done by my eclipse formatter using the formatter file provided on tachyon webpage.

File: core/src/main/java/tachyon/client/BlockOutStream.java (hunk 1 above)
Comment (apc999) [inline core/src/main/java/tachyon/client/BlockOutStream.java]: When I was using
"
if (mBuffer.position() >=     mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES,		 
        Constants.MB)) {
"
and run "mvn package -DskipTests",
I got this "/apc999/core/src/main/java/tachyon/client/BlockOutStream.java:235:28: '>=' should be on a new line."

So I moved ">=" down to make style-checker happy and let me proceed.

Comment (apc999) [inline core/src/main/java/tachyon/client/BlockOutStream.java]: sorry please ignore my previous post. This was done by my eclipse formatter (i didn't touch it)

File: core/src/main/java/tachyon/client/TachyonFile.java (hunk 2 above)
Comment (hsaputra) [inline core/src/main/java/tachyon/client/TachyonFile.java]: Ah ok, thx!

**Pull request merged**